### PR TITLE
Introduce authors

### PR DESCRIPTION
**Pull request by mawit**
With this PR it is possible to enter an author for a post.

New fields for frontmatter:

```
author:
  link: /category/dexter
  name: Dexter
  gravatar: 2bfa103a13c88b5ffd26da6f982f11df
```

Mandatory fields (if `author` is added):
* `name` (The full name/display name of the author)
* `link` (Link to details about the author or a post list. For the post list I would recommend using a category with the same name as the author)

`gravatar` is not mandatory. It is simply not shown if one does not specify any. It is also possible to define a default gravatar for the whole blog in `themeConfig.defaultAuthorGravatar`.
I use only the hashes there to not disclose email addresses.

If `author` is not present in frontmatter there is not author shown for the post.

### CODE REVIEW

**Comment by alexander-heimbuch** [comment]
Thanks for the contributions, I will have a look this weekend to get them merged and released!

File: blog/.vuepress/theme/layouts/Post.vue [hunk 1]
```diff
@@ -13,7 +13,18 @@
 
           <div class="post-full-byline">
             <section class="post-full-byline-content">
+              <ul class="author-list" v-if="current.author && gravatar">
```
Comment (alexander-heimbuch) [inline blog/.vuepress/theme/layouts/Post.vue:16]: I would suggest to have a `author` as a store value and extract the logic `current.author && gravatar`.

File: blog/.vuepress/theme/layouts/Post.vue [hunk 2]
```diff
@@ -77,6 +88,16 @@ export default {
       return {
         "background-image": `url(${this.$withBase(this.current.image)})`
       };
+    },
+
+    gravatar () {
```
Comment (alexander-heimbuch) [inline blog/.vuepress/theme/layouts/Post.vue:93]: I would suggest to move this logic to `https://github.com/alexander-heimbuch/vuepress-theme-casper/blob/master/blog/.vuepress/theme/store/utils.js#L4`

**Comment by mawit** [comment]
I have tried to fix this in https://github.com/alexander-heimbuch/vuepress-theme-casper/pull/22/commits/34dedb5b5459bde75cfb323817ead5e73de7814d but unsure if this is what you meant.

**Comment by alexander-heimbuch** [comment]
closed in favour for #24

**Pull request closed without merging**